flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/FileSystemJobResultStoreTest.java

```diff
@@ -0,0 +1,285 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.highavailability;
+
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.core.fs.Path;
+import org.apache.flink.runtime.clusterframework.ApplicationStatus;
+import org.apache.flink.runtime.jobmaster.JobResult;
+import org.apache.flink.util.TestLoggerExtension;
+
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+
+/** Tests for the {@link FileSystemJobResultStore}. */
+@ExtendWith(TestLoggerExtension.class)
+public class FileSystemJobResultStoreTest {
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    private static final JobResultEntry DUMMY_JOB_RESULT_ENTRY = createDummyJobResultEntry();
+
+    private FileSystemJobResultStore fileSystemJobResultStore;
+
+    @TempDir File temporaryFolder;
+
+    @BeforeEach
+    public void setupTest() throws IOException {
+        Path path = new Path(temporaryFolder.toURI());
+        fileSystemJobResultStore = new FileSystemJobResultStore(path.getFileSystem(), path, false);
+    }
+
+    /** Tests that the base directory is created on Filesystem Job Result Store initialization. */
+    @Test
+    public void testBaseDirectoryCreationOnResultStoreInitialization() throws Exception {
+        final File tempDir = new File(temporaryFolder.getPath() + "/new-temp-dir");
```

Comment:
```suggestion
        final File tempDir = new File(temporaryFolder.getPath(), "new-temp-dir");
```
nit: to avoid this separator specification